### ISSUE TITLE
Support failed allocation metrics in 2.2.0

### ISSUE THREAD

**Issue by mweinelt** [mweinelt/kea-exporter#25]
```
2. **New statistics**: New statistics were have been added to kea-dhcp4 
to counts cases of host reservation conflicts. They are tracked at both 
the global and subnet level as `v4-reservation-conflicts` and 
`subnet[id].v4-reservation-conflicts`, respectively [#2419].
```

```
9. **New statistics for failed allocations**. If the class requirements 
for your address pools are defined too tightly, it is possible that some 
clients will not get an address. To ease the investigation of this 
problem, many new statistics were added:
`v4-allocation-fail`, `v4-allocation-fail-shared-network`, 
`v4-allocation-fail-subnet`, `v4-allocation-fail-no-pools`, 
`v4-allocation-fail-classes`, `subnet[X].v4-allocation-fail`, 
`subnet[X].v4-allocation-fail-shared-network`, 
`subnet[X].v4-allocation-fail-subnet`, 
`subnet[X].v4-allocation-fail-no-pools`, 
`subnet[X].v4-allocation-fail-classes`,
`v6-allocation-fail`, `v6-allocation-fail-shared-network`, 
`v6-allocation-fail-subnet`, `v6-allocation-fail-no-pools`, 
`v6-allocation-fail-classes`, `subnet[X].v6-allocation-fail`, 
`subnet[X].v6-allocation-fail-shared-network`, 
`subnet[X].v6-allocation-fail-subnet`, 
`subnet[X].v6-allocation-fail-no-pools`, and 
`subnet[X].v6-allocation-fail-classes` [#2054].
```

https://downloads.isc.org/isc/kea/2.2.0/Kea-2.2.0-ReleaseNotes.txt

**Comment by glendekoning** [comment]
Heya @mweinelt, i think i have covered everything. let me know if you think i should make any changes.

**Comment by mweinelt** [comment]
In the meantime I've released version 0.4.4 that fails much more gracefully in the face of unknown metrics.
